content/docs/guides/testing/unit.md

```diff
@@ -330,18 +330,22 @@ pulumi.runtime.setMocks({
     // ... mocks as shown above
 });
 
-// It's important to import the program _after_ the mocks are defined.
-import * as infra from "./index";
-
 describe("Infrastructure", function() {
-    const server = infra.server;
+    let infra: typeof import('../index');
+
+    before(async function() {
+        // It's important to import the program _after_ the mocks are defined.
+        infra = await import('../index');
+    })
+
     describe("#server", function() {
+        const server = infra.server;
```

Comment:
looking back, this is broken. We cannot destructure inside `describe` as that runs before any hooks (like `before`) and tests. It needs to be accessed either after the `await` or inside a test

Reply:
(at least that's the way Jest and Jasmine works, I don't think I've every used Mocha. But I assume it also works that way)